pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexCRUDViolationRule.java

```diff
@@ -351,11 +369,33 @@ private void checkForAccessibility(final AbstractApexNode<?> node, Object data)
 
 			if (varToTypeMapping.containsKey(variableWithClass.toString())) {
 				String type = varToTypeMapping.get(variableWithClass.toString());
+				if (!isGetter) {
+					validateCRUDCheckPresent(node, data, ANY, type);
+				}
+			}
 
-				validateCRUDCheckPresent(node, data, ANY, type);
+		}
 
+		final ASTReturnStatement returnStatement = node.getFirstParentOfType(ASTReturnStatement.class);
+		if (returnStatement != null) {
+			if (!isGetter) {
+				validateCRUDCheckPresent(node, data, ANY, returnType == null ? "" : returnType);
 			}
-
 		}
+
+	}
+
+	private String getReturnType(final ASTMethod method) {
+		return new StringBuilder().append(method.getNode().getDefiningType().getApexName()).append(":")
+				.append(method.getNode().getMethodInfo().getEmitSignature().getReturnType().getApexName()).toString();
+	}
+
+	private boolean isMethodAGetter(final ASTMethod method) {
+		final Pattern p = Pattern.compile("^(string|void)$", Pattern.CASE_INSENSITIVE);
```

Comment:
compiling a `Pattern` is expensive. Since this is a constant string, you should move this to a `static final` field.